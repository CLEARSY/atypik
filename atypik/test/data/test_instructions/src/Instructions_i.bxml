<?xml version="1.0" encoding="UTF-8"?>
<Machine xmlns='https://www.atelierb.eu/Formats/bxml' version='1.0' name='Instructions_i'
    type='implementation' position='true' semantic='true' b0check='false'>
    <Abstraction>Instructions_r</Abstraction>
    <Imports>
        <Attr>
            <Pos l='9' c='1' s='7' />
        </Attr>
        <Referenced_Machine>
            <Attr>
                <Pos l='10' c='5' s='14' />
            </Attr>
            <Name>Operations</Name>
            <Instance>lib</Instance>
        </Referenced_Machine>
    </Imports>
    <Values>
        <Attr>
            <Pos l='12' c='1' s='6' />
        </Attr>
        <Valuation ident='ABS_SET' typref='2'>
            <Attr>
                <Pos l='13' c='5' s='7' />
            </Attr>
            <Binary_Exp op='..' typref='2'>
                <Attr>
                    <Pos l='13' c='16' s='2' />
                </Attr>
                <Integer_Literal value='1' typref='1'>
                    <Attr>
                        <Pos l='13' c='15' s='1' />
                    </Attr>
                </Integer_Literal>
                <Integer_Literal value='10' typref='1'>
                    <Attr>
                        <Pos l='13' c='18' s='2' />
                    </Attr>
                </Integer_Literal>
            </Binary_Exp>
        </Valuation>
    </Values>
    <Concrete_Variables>
        <Attr>
            <Pos l='15' c='1' s='18' />
        </Attr>
        <Id value='var_i' typref='1'>
            <Attr>
                <Pos l='16' c='5' s='5' />
            </Attr>
        </Id>
        <Id value='var_i2' typref='1'>
            <Attr>
                <Pos l='16' c='12' s='6' />
            </Attr>
        </Id>
        <Id value='var_i3' typref='1'>
            <Attr>
                <Pos l='16' c='20' s='6' />
            </Attr>
        </Id>
    </Concrete_Variables>
    <Invariant>
        <Attr>
            <Pos l='18' c='1' s='9' />
        </Attr>
        <Nary_Pred op='&amp;'>
            <Exp_Comparison op=':'>
                <Attr>
                    <Pos l='19' c='11' s='1' />
                </Attr>
                <Id value='var_i' typref='1'>
                    <Attr>
                        <Pos l='19' c='5' s='5' />
                    </Attr>
                </Id>
                <Id value='INT' typref='2'>
                    <Attr>
                        <Pos l='19' c='13' s='3' />
                    </Attr>
                </Id>
            </Exp_Comparison>
            <Exp_Comparison op=':'>
                <Attr>
                    <Pos l='20' c='12' s='1' />
                </Attr>
                <Id value='var_i2' typref='1'>
                    <Attr>
                        <Pos l='20' c='5' s='6' />
                    </Attr>
                </Id>
                <Id value='INT' typref='2'>
                    <Attr>
                        <Pos l='20' c='14' s='3' />
                    </Attr>
                </Id>
            </Exp_Comparison>
            <Exp_Comparison op=':'>
                <Attr>
                    <Pos l='21' c='12' s='1' />
                </Attr>
                <Id value='var_i3' typref='1'>
                    <Attr>
                        <Pos l='21' c='5' s='6' />
                    </Attr>
                </Id>
                <Id value='INT' typref='2'>
                    <Attr>
                        <Pos l='21' c='14' s='3' />
                    </Attr>
                </Id>
            </Exp_Comparison>
        </Nary_Pred>
    </Invariant>
    <Initialisation>
        <Attr>
            <Pos l='23' c='1' s='14' />
        </Attr>
        <Nary_Sub op=';'>
            <VAR_IN>
                <Attr>
                    <Pos l='24' c='5' s='3' />
                </Attr>
                <Variables>
                    <Id value='x1' typref='1'>
                        <Attr>
                            <Pos l='24' c='9' s='2' />
                        </Attr>
                    </Id>
                    <Id value='x2' typref='1'>
                        <Attr>
                            <Pos l='24' c='13' s='2' />
                        </Attr>
                    </Id>
                </Variables>
                <Body>
                    <Nary_Sub op=';'>
                        <Assignement_Sub>
                            <Attr>
                                <Pos l='25' c='12' s='2' />
                            </Attr>
                            <Variables>
                                <Id value='x1' typref='1'>
                                    <Attr>
                                        <Pos l='25' c='9' s='2' />
                                    </Attr>
                                </Id>
                            </Variables>
                            <Values>
                                <Integer_Literal value='2' typref='1'>
                                    <Attr>
                                        <Pos l='25' c='15' s='1' />
                                    </Attr>
                                </Integer_Literal>
                            </Values>
                        </Assignement_Sub>
                        <Assignement_Sub>
                            <Attr>
                                <Pos l='26' c='12' s='2' />
                            </Attr>
                            <Variables>
                                <Id value='x2' typref='1'>
                                    <Attr>
                                        <Pos l='26' c='9' s='2' />
                                    </Attr>
                                </Id>
                            </Variables>
                            <Values>
                                <Binary_Exp op='*i' typref='1'>
                                    <Attr>
                                        <Pos l='26' c='18' s='1' />
                                    </Attr>
                                    <Id value='x1' typref='1'>
                                        <Attr>
                                            <Pos l='26' c='15' s='2' />
                                        </Attr>
                                    </Id>
                                    <Integer_Literal value='2' typref='1'>
                                        <Attr>
                                            <Pos l='26' c='20' s='1' />
                                        </Attr>
                                    </Integer_Literal>
                                </Binary_Exp>
                            </Values>
                        </Assignement_Sub>
                        <Assignement_Sub>
                            <Attr>
                                <Pos l='27' c='15' s='2' />
                            </Attr>
                            <Variables>
                                <Id value='var_i' typref='1'>
                                    <Attr>
                                        <Pos l='27' c='9' s='5' />
                                    </Attr>
                                </Id>
                            </Variables>
                            <Values>
                                <Binary_Exp op='+i' typref='1'>
                                    <Attr>
                                        <Pos l='27' c='21' s='1' />
                                    </Attr>
                                    <Id value='x1' typref='1'>
                                        <Attr>
                                            <Pos l='27' c='18' s='2' />
                                        </Attr>
                                    </Id>
                                    <Id value='x2' typref='1'>
                                        <Attr>
                                            <Pos l='27' c='23' s='2' />
                                        </Attr>
                                    </Id>
                                </Binary_Exp>
                            </Values>
                        </Assignement_Sub>
                    </Nary_Sub>
                </Body>
            </VAR_IN>
            <Operation_Call>
                <Attr>
                    <Pos l='29' c='20' s='3' />
                </Attr>
                <Name>
                    <Id value='lib.pair' instance='lib' component='pair'>
                        <Attr>
                            <Pos l='29' c='24' s='8' />
                        </Attr>
                    </Id>
                </Name>
                <Input_Parameters>
                    <Integer_Literal value='0' typref='1'>
                        <Attr>
                            <Pos l='29' c='33' s='1' />
                        </Attr>
                    </Integer_Literal>
                    <Integer_Literal value='1' typref='1'>
                        <Attr>
                            <Pos l='29' c='36' s='1' />
                        </Attr>
                    </Integer_Literal>
                </Input_Parameters>
                <Output_Parameters>
                    <Id value='var_i2' typref='1'>
                        <Attr>
                            <Pos l='29' c='5' s='6' />
                        </Attr>
                    </Id>
                    <Id value='var_i3' typref='1'>
                        <Attr>
                            <Pos l='29' c='13' s='6' />
                        </Attr>
                    </Id>
                </Output_Parameters>
            </Operation_Call>
        </Nary_Sub>
    </Initialisation>
    <TypeInfos>
        <Type id='0'>
            <Id value='BOOL' />
        </Type>
        <Type id='1'>
            <Id value='INTEGER' />
        </Type>
        <Type id='2'>
            <Unary_Exp op='POW'>
                <Id value='INTEGER' />
            </Unary_Exp>
        </Type>
        <Type id='3'>
            <Unary_Exp op='POW'>
                <Unary_Exp op='POW'>
                    <Id value='INTEGER' />
                </Unary_Exp>
            </Unary_Exp>
        </Type>
    </TypeInfos>

</Machine>