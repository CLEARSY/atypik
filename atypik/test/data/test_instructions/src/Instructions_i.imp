/* Instructions_i
* Author: vdemeulenaere
* Creation date: 13/06/2023
*/

IMPLEMENTATION Instructions_i
REFINES Instructions_r

IMPORTS
    lib.Operations
    
VALUES 
    ABS_SET = 1..10
    
CONCRETE_VARIABLES
    var_i, var_i2, var_i3
    
INVARIANT
    var_i : INT &
    var_i2 : INT &
    var_i3 : INT
    
INITIALISATION
    VAR x1, x2 IN
        x1 := 2;
        x2 := x1 * 2;
        var_i := x1 + x2
    END;
    var_i2, var_i3 <-- lib.pair(0, 1)
    
    
END