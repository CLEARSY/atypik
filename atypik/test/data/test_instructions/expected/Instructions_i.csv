Expression	Identifier	Line	Column	Type
"1"	59	29	36	INTEGER
"lib.pair"	57	29	24	(POW (PRODUCT (PRODUCT (PRODUCT INTEGER INTEGER) INTEGER) INTEGER))
"(x1) +i (x2)"	56	27	21	INTEGER
"(x1) *i (2)"	55	26	18	INTEGER
"2"	54	26	20	INTEGER
"x2"	52	27	23	INTEGER
"x1"	51	24	9	INTEGER
"ABS_SET = (1) .. (10)"	47	13	5	(POW INTEGER)
"var_i2"	49	29	5	INTEGER
"(1) .. (10)"	46	13	16	(POW INTEGER)
"10"	45	13	18	INTEGER
"var_i3"	50	16	20	INTEGER
"var_i"	48	27	9	INTEGER
"0"	58	29	33	INTEGER
"2"	53	25	15	INTEGER
"1"	44	13	15	INTEGER
